src/MongoDB.Driver/Linq/Translators/PredicateTranslator.cs

```diff
@@ -1654,11 +1666,6 @@ protected internal override Expression VisitDocument(DocumentExpression node)
             {
                 return new FieldExpression("", node.Serializer);
             }
-
-            protected internal override Expression VisitPipeline(PipelineExpression node)
-            {
-                return node;
-            }
```

Comment:
@rstam there is the difference in this potential(new) test:
`c => !c.M.Any(a => !(x.Contains(a) && x.Contains(a)))`
if we have the code from this point we will have throwing the unsuppotted exception on driver side with the message: `Unsupported filter: Contains(value(System.Int32[])).`.

But if we delete this code, the generated query will look like:
`{ "M\" : { "$not\" : { "$elemMatch\" : { "$nor" : [{ "$and" : [{ "$in" : [1, 2] }, { "$in" : [1, 2] }] }] } } } }`
Then, the query will throw exception on SERVER side:
`Command find failed: unknown top level operator: $in.`

Reply:
That seems like the only possible translation for that LINQ query.

While it is true that the server does not currently support that query, it COULD support it. The current limitation on the server is that only supports implied left hand side for `{ $in : [...] }` at the top level. Once inside an `$and` the server seems to only support named left hand side (e.g. `{ name : { $in : [...] } }`

Reply:
Yep, but I guess having an exception on the DRIVER side better than on SERVER? As soon as the SERVER will support this query we can think to fix this part. And I guess that time we will need to think about other possible tests.